pkg/experiment/metastore/dlq/recovery.go

```diff
@@ -91,62 +91,71 @@ func (r *Recovery) recoverLoop(ctx context.Context) {
 }
 
 func (r *Recovery) recoverTick(ctx context.Context) {
-	err := r.bucket.Iter(ctx, block.DirNameDLQ, func(metaPath string) error {
-		if ctx.Err() != nil {
-			return ctx.Err()
-		}
-		return r.recover(ctx, metaPath)
+	err := r.bucket.Iter(ctx, block.DirNameDLQ, func(path string) error {
+		return r.recover(ctx, path)
 	}, objstore.WithRecursiveIter)
```

Comment:
This is not helpful to check the context in this way. If the iterator doesn't handle context errors (implementations we use do handle it well), it's a bug.